custom_components/thermal_comfort/sensor.py

```diff
@@ -237,14 +240,73 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     return True
 
 
-class SensorThermalComfort(SensorEntity):
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up entity configured via user interface.
+
+    Called via async_setup_platforms(, SENSOR) from __init__.py
+    """
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    enabled_sensor_types: list[SensorType] = []
+    _LOGGER.debug(f"async_setup_entry: {data}")
+    for c in SensorType:
+        if data.get(c, False):
+            enabled_sensor_types.append(c)
+
+    compute_device = DeviceThermalComfort(
+        hass,
+        data[CONF_TEMPERATURE_SENSOR],
+        data[CONF_HUMIDITY_SENSOR],
+        SENSOR_TYPES,
+        data[CONF_POLL],
+    )
+
+    entities: list[SensorThermalComfortEntry] = []
+    for sensor_type in SENSOR_TYPES:
+        entity_description = SensorEntityDescription(**SENSOR_TYPES[sensor_type])
+        entity_description.entity_registry_enabled_default = (
+            True if sensor_type in enabled_sensor_types else False
+        )
+
+        _LOGGER.debug(
+            f"async_setup_entry: setting up entity for {sensor_type},"
+            f"enabled_default = {entity_description.entity_registry_enabled_default}"
+        )
+        entities.append(
+            SensorThermalComfortEntry(
+                device=compute_device,
+                entity_description=entity_description,
+                hass=hass,
+                sensor_type=sensor_type,
+                device_name=data[CONF_NAME],
+                device_unique_id=f"{config_entry.unique_id}",
+            )
+        )
+    if entities:
+        async_add_entities(entities)
+        for e in entities:
+            e.async_schedule_update_ha_state(force_refresh=True)
```

Comment:
This does what you intended and updates them after creation but errors out on server restart since it is called to early.
Remove this lines and change `async_addet_to_hass` of `SensorThermalComfortCommon` to:
```python
    async def async_added_to_hass(self):
        """Register callbacks."""
        self._device.sensors.append(self)
        if self._icon_template is not None:
            self._icon_template.hass = self.hass
        if self._entity_picture_template is not None:
            self._entity_picture_template.hass = self.hass
        if self._device._compute_states[self._sensor_type].needs_update:
            self.async_schedule_update_ha_state(True)
```
basically adding the lines
```python
        if self._device._compute_states[self._sensor_type].needs_update:
            self.async_schedule_update_ha_state(True)
```

And change the `ComputeState` the default value of `needs_update` to `False`

Reply:
We may call just 
```python
async def async_added_to_hass(self):
...
    self._device.async_update()
    self.async_schedule_update_ha_state(True)
```
It will be equivalent, without changing defaults. Will it better?

Reply:
Calling `self._device.async_update()` would retrigger all sensors that are already added to the device. We just want to update the sensor that has just been added not those that have been added previously. This should only happen if both input sensors have already set a valid value (device._temperature, device._humidity).